src/shared_memory.h

```diff
@@ -1,11 +1,27 @@
 #pragma once
-#include <array>
-#include <cstdio>
+
 #include "common_types.h"
 
 namespace Teakra {
 struct SharedMemory {
-    std::array<u8, 0x80000> raw{};
+    u8* raw;
+    bool own_dsp_mem;
```

Comment:
I think a better way in modern C++ to do this is something like
```
std::option<std::unique_ptr<std::array<u8, 0x80000>>> raw;
u8* raw_ptr;
```
so when Teakra manages the memory, it doesn't need to manually do `new`/`delete`, and it reduce the chance we mishandle and leak it